service/resource_mapper.go

```diff
@@ -90,22 +90,36 @@ func (m *resourceMapper) MapResource(ctx context.Context, spec authz.SubjectAcce
 				"----%s's request is not allowed----\nVerb:\t%s\nNamespaceb:\t%s\nAPI Group:\t%s\nResource:\t%s\nResource Name:\t%s\n",
 				identity, verb, namespace, group, resource, name)
 	case m.res.IsAdminAccess(verb, namespace, group, resource, name):
-		return identity, []webhook.AthenzAccessCheck{
-			{
+		return identity, m.createAdminAcessCheck(adminDomain, group, resource, name, verb, domains), nil
+	default:
+		return identity, m.createAcessCheck(group, resource, name, verb, domains), nil
+	}
+}
+
+func (m *resourceMapper) createAdminAcessCheck(adminDomain, group, resource, name, verb string, domains []string) []webhook.AthenzAccessCheck {
+	accessChecks := make([]webhook.AthenzAccessCheck, 0, len(domains)*2)
+	for _, domain := range domains {
+		accessChecks = append(accessChecks,
+			webhook.AthenzAccessCheck{
 				Resource: m.res.TrimResource(fmt.Sprintf("%s:%s.%s.%s.%s", adminDomain, group, domain, resource, name)),
 				Action:   verb,
 			},
-			{
+			webhook.AthenzAccessCheck{
 				Resource: m.res.TrimResource(fmt.Sprintf("%s:%s.%s.%s", adminDomain, group, resource, name)),
 				Action:   verb,
-			},
-		}, nil
-	default:
-		return identity, []webhook.AthenzAccessCheck{
-			{
+			})
```

Comment:
this request is duplicated? what it the correct order to check the adminDomain only?
`[ adminDomain+domain_0, adminDomain, adminDomain+domain_1, adminDomain, ... ]`

Reply:
I think this code is implemented according to [this](https://github.com/yahoojapan/garm/blob/master/doc/garm-functional-overview.md#filter-k8s-request) document.

> Athenz admin domain (2 requests to Athenz, OR logic, any one is allowed implies the action is allowed.) 

Consistent with this description, making requests with and without a domain.

Reply:
I understand.
It means that adminDomain requests are duplicated.
I will fix it.

Reply:
Fixed 3720724 .
Thank you.